rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java

```diff
@@ -884,6 +890,53 @@ public void saveReceiveHeadersLastTimestamp() {
         }
     }
 
+    public Optional<Long> getUtxoExpirationCheckpointRskBlockHeight() {
+        if (!activations.isActive(RSKIP264)) {
+            return Optional.empty();
+        }
+
+        if (utxoExpirationCheckpointRskBlockHeight == null) {
+            utxoExpirationCheckpointRskBlockHeight = safeGetFromRepository(UTXO_EXPIRATION_CHECK_POINT_RSK_BLOCK_HEIGHT, BridgeSerializationUtils::deserializeOptionalLong).orElse(0L);
+        }
+
+        return Optional.of(utxoExpirationCheckpointRskBlockHeight);
+    }
+
+    public void setUtxoExpirationCheckpointRskBlockHeight(long utxoExpirationCheckpointRskBlockHeight) {
+        if (activations.isActive(RSKIP264)) {
+            this.utxoExpirationCheckpointRskBlockHeight = utxoExpirationCheckpointRskBlockHeight;
+        }
+    }
+
+    protected void saveUtxoExpirationCheckpointRskBlockHeight() {
+        if (activations.isActive(RSKIP264) && this.utxoExpirationCheckpointRskBlockHeight != null) {
+            safeSaveToRepository(UTXO_EXPIRATION_CHECK_POINT_RSK_BLOCK_HEIGHT, this.utxoExpirationCheckpointRskBlockHeight, BridgeSerializationUtils::serializeLong);
+        }
+    }
+
+    public Optional<Long> getUtxoExpirationCheckpointTimestamp() {
+        if (!activations.isActive(RSKIP264)) {
+            return Optional.empty();
+        }
+
+        if (utxoExpirationCheckpointTimestamp == null) {
+            utxoExpirationCheckpointTimestamp = safeGetFromRepository(UTXO_EXPIRATION_CHECK_POINT_TIMESTAMP, BridgeSerializationUtils::deserializeOptionalLong).orElse(0L);
+        }
+        return Optional.of(utxoExpirationCheckpointTimestamp);
+    }
+
+    public void setUtxoExpirationCheckpointTimestamp(Long utxoExpirationCheckpointTimestamp) {
+        if (activations.isActive(RSKIP264)) {
+            this.utxoExpirationCheckpointTimestamp = utxoExpirationCheckpointTimestamp;
+        }
+    }
+
+    protected void saveUtxoExpirationCheckpointTimestamp() {
+        if (activations.isActive(RSKIP264) && this.utxoExpirationCheckpointTimestamp > 0) {
```

Comment:
I would move the checking of timestamp > 0 to the setter method. Here we are only trying to save the value we already have stored in memory

Reply:
Makes Sense. Resolved